processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -542,7 +543,7 @@ private static void setupDimConversion(
 
       for (int i = 0; i < adapters.size(); i++) {
         Indexed<String> dimValues = adapters.get(i).getDimValueLookup(dimension);
-        if (dimValues != null) {
+        if (!IndexMerger.isNullColumn(dimValues)) {
```

Comment:
dimValues returned by IndexableAdapter should always return null, so there should not be any reason to check for empty string in isNullColumn